examples/pip_parse/WORKSPACE

```diff
@@ -41,6 +41,19 @@ pip_parse(
     # Here, we use the interpreter constant that resolves to the host interpreter from the default Python toolchain.
     python_interpreter_target = interpreter,
 
+    # Requirement groups allow Bazel to tolerate PyPi cycles by putting dependencies
+    # which are known to form cycles into groups together.
+    requirement_groups = {
+        "airflow": [
+            "apache-airflow",
+            "apache-airflow-providers-common-sql",
+            "apache-airflow-providers-ftp",
+            "apache-airflow-providers-http",
+            "apache-airflow-providers-imap",
+            "apache-airflow-providers-sqlite",
+        ],
```

Comment:
I think using `airflow` as an example for this is not great. The reason why the providers are linking back to a particular version of `airflow` is for compatibility reasons. This dependency is used when doing version resolution (i.e. a particular version of an `apache-airflow-provider` is incompatible with older `airflow` versions) and they don't have actual import cycles, so it is perfectly fine to cut the dependency cycle by patching out the `airflow` dependency in the providers. 

That said, lumping them all into one group also makes sense conceptually, because even after patching things out, we will always have the list of packages present together.